dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -62,6 +75,42 @@ public fun DidIonManager(builderAction: DidIonConfiguration.() -> Unit): DidIonM
 /** [DidIonManager] is sealed, so we provide an impl so the constructor can be called. */
 private class DidIonManagerImpl(configuration: DidIonConfiguration) : DidIonManager(configuration)
 
+/**
+ * The options when updating an ION did.
+ *
+ * @param didString The full did. e.g. `did:ion:123`
+ * @param updateKeyAlias The alias within the key manager that refers to the last update key.
+ * @param servicesToAdd The services to add in the did document.
+ * @param idsOfServicesToRemove Ids of the services to remove from the did document.
+ * @param publicKeysToAdd PublicKeys to add to the DID document.
+ * @param idsOfPublicKeysToRemove Keys to remove from the DID document.
+ */
+public class UpdateDidIonOptions(
```

Comment:
A possible alternative to this impl might be

```
public data class UpdateDidIonOptions(
  val didString: String,
  val updateKeyAlias: String,
  val servicesToAdd: List<Service> = emptyList(),
  val idsOfServicesToRemove: List<String> = emptyList(),
  val publicKeysToAdd: List<PublicKey> = emptyList(),
  val idsOfPublicKeysToRemove: List<String> = emptyList()
) {
  internal fun toPatches(): List<PatchAction> {
    fun <T> MutableList<PatchAction>.addIfNotEmpty(list: List<T>, action: (List<T>) -> PatchAction) {
      list.takeIf { it.isNotEmpty() }?.let { this.add(action(it)) }
    }

    return buildList {
      addIfNotEmpty(servicesToAdd, ::AddServicesAction)
      addIfNotEmpty(idsOfServicesToRemove, ::RemoveServicesAction)
      addIfNotEmpty(publicKeysToAdd, ::AddPublicKeysAction)
      addIfNotEmpty(idsOfPublicKeysToRemove, ::RemovePublicKeysAction)
    }
  }
}
```
For a rough and untested stab at it. I think this covers the comments below.

(https://github.com/TBD54566975/web5-kt/pull/77#discussion_r1354623761)